src/mds/Server.cc

```diff
@@ -598,7 +598,11 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
 
       if (blocklisted) {
 	dout(10) << "rejecting blocklisted client " << addr << dendl;
-	send_reject_message("blocklisted");
+	// The "blacklisted" string goes on the wire and is depended
+	// upon by the kernel client for detecting whether it has been
+	// blocklisted.  If mounted with recover_session=clean (since
+	// 5.4), it tries to automatically recover from blocklisting.
+	send_reject_message("blacklisted");
```

Comment:
We can also do:

```suggestion
	send_reject_message("blacklisted|blocklisted");
```

Then patch the kernel to accept blocklisted. Other clients can use "blocklisted" for detection. Someday in the far future maybe we'll remove `blacklisted|`.

Reply:
Generally, when it comes to backwards compatibility bugs such as this one, I prefer just reverting because you never know what is out there in the wild.  For example, if the kernel client did `strcmp`, revert would be the only sane option.  But since it does substring search and presumably no other clients exist, we can do that (or maybe `"blocklisted (blacklisted)"` so that it is more human readable and the new adjective is first).

Reply:
`"blocklisted (blacklisted)"` is also acceptable. I'd prefer we move in a direction where `blacklisted` can one day be removed. We're indeed fortunate that `strnstr` was used and not `strcmp`.

Reply:
Yup, we need to stop parsing this string in the kernel client.  Making random strings part of the on-wire format is bad regardless of their content...

Reply:
I agree but that's where we are. A tracker ticket to fix this by adding a flag to MClientSession would be appropriate. I agree that the "error_string" in the session metadata should be for humans.

Reply:
https://tracker.ceph.com/issues/47450